packages/hub/routes/email-card-drop-requests.ts

```diff
@@ -0,0 +1,54 @@
+import Koa from 'koa';
+import autoBind from 'auto-bind';
+import { query } from '../queries';
+import { inject } from '@cardstack/di';
+
+export interface EmailCardDropRequest {
+  id: string;
+  ownerAddress: string;
+  emailHash: string;
+  verificationCode: string;
+  claimedAt?: Date;
+  requestedAt: Date;
+  transactionHash?: string;
+}
+
+export default class EmailCardDropRequestsRoute {
+  emailCardDropRequestQueries = query('email-card-drop-requests', { as: 'emailCardDropRequestQueries' });
+  emailCardDropRequestSerializer = inject('email-card-drop-request-serializer', {
+    as: 'emailCardDropRequestSerializer',
+  });
+  clock = inject('clock');
+
+  constructor() {
+    autoBind(this);
+  }
+
+  async get(ctx: Koa.Context) {
+    let timestamp = new Date(this.clock.now());
+    let ownerAddress = ctx.request.query['eoa'] as string;
+
+    let previousRequests = await this.emailCardDropRequestQueries.query({
+      ownerAddress,
+    });
```

Comment:
could it be a problem  if `eoa` is undefined ?

Reply:
Response body will now be:

```
{
      errors: [
        {
          code: '400',
          title: 'Missing required parameter: eoa',
          detail: 'Please provide an ethereum address via the eoa query parameter',
        },
      ],
}
```

for missing EOA